src/static/helpers/slasHelper.ts

```diff
@@ -0,0 +1,258 @@
+    redirectURI: string;
+    usid?: string;
+  }
+): Promise<TokenResponse> {
+  const codeVerifier = createCodeVerifier();
+
+  const authResponse = await authorize(slasClient, {
+    codeVerifier,
+    redirectURI: parameters.redirectURI,
+    hint: 'guest',
+    ...(parameters.usid && {usid: parameters.usid}),
+  });
+
+  const tokenBody: TokenRequest = {
+    client_id: slasClient.clientConfig.parameters.clientId,
+    code: authResponse.code,
+    code_verifier: codeVerifier,
+    grant_type: 'authorization_code_pkce',
+    redirect_uri: parameters.redirectURI,
+    usid: authResponse.usid,
+  };
+
+  return slasClient.getAccessToken({body: tokenBody});
+}
+
+/**
+ * @param slasClient a configured instance of the ShopperLogin SDK client
+ * @param parameters -
+ *  shopperUserId: the id of the user to login with;
+ *  shopperPassword: the password of the user to login with;
+ *  redirectURI: a registered redirect URI to return;
+ *  usid?: an existing usid to continue a session;
+ * @returns TokenResponse object
+ */
+export async function loginRegisteredUserB2C(
+  slasClient: ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+    siteId: string;
+  }>,
+  parameters: {
+    shopperUserId: string;
+    shopperPassword: string;
+    redirectURI: string;
+    usid?: string;
+  }
+): Promise<TokenResponse> {
+  const codeVerifier = createCodeVerifier();
+  const codeChallenge = await generateCodeChallenge(codeVerifier);
+
+  const authorization = `Basic ${btoa(
+    `${parameters.shopperUserId}:${parameters.shopperPassword}`
+  )}`;
+
+  const options = {
+    headers: {
+      Authorization: authorization,
+    },
+    parameters: {
+      organizationId: slasClient.clientConfig.parameters.organizationId,
+    },
+    body: {
+      redirect_uri: parameters.redirectURI,
+      client_id: slasClient.clientConfig.parameters.clientId,
+      code_challenge: codeChallenge,
+      channel_id: slasClient.clientConfig.parameters.siteId,
+      ...(parameters.usid && {usid: parameters.usid}),
+    },
+  };
+
+  const response = await slasClient.authenticateCustomer(options, true);
+
+  const authResponse = getCodeAndUsidFromUrl(response.url);
+
+  const tokenBody = {
+    client_id: slasClient.clientConfig.parameters.clientId,
+    code: authResponse.code,
+    code_verifier: codeVerifier,
+    grant_type: 'authorization_code_pkce',
+    organizationId: slasClient.clientConfig.parameters.organizationId,
+    redirect_uri: parameters.redirectURI,
+    usid: authResponse.usid,
+  };
+
+  return slasClient.getAccessToken({body: tokenBody});
+}
+
+/**
+ * Exchange a refresh token for a new access token.
+ * @param slasClient a configured instance of the ShopperLogin SDK client
+ * @param slasRefreshToken
+ * @returns
+ */
+export function refreshToken(
+  slasClient: ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+  }>,
+  parameters: {refreshToken: string}
```

Comment:
I'd agree, but I'm preferring to leave it because 'refresh' being a noun and verb here means you have a clash between the function name and parameter name. Since keeping `parameters` is consist with the other functions *and* prevents the name clash, it gets my vote

Reply:
That's a good point, though, that you need a refresh token to refresh the access token. Could the method name be more clear?

Reply:
I'm struggling to come up with it myself, but is `refreshAccessToken` clearer maybe?